src/Control/Carrier/Interpret.hs

```diff
@@ -63,24 +62,24 @@ reify a k = unsafeCoerce (Magic k) a
 --
 -- @since 1.0.0.0
 runInterpret
-  :: (forall n x . Monad n => (forall y . n y -> m y) -> eff n x -> m x)
-  -> (forall s . Reifies s (Handler eff m) => InterpretC s eff m a)
+  :: (forall ctx n x . Functor ctx => Handler ctx n m -> eff n x -> ctx () -> m (ctx x))
+  -> (forall s . Reifies s (Interpreter eff m) => InterpretC s eff m a)
   -> m a
-runInterpret f m = reify (Handler (\ hom -> InterpretC . f (runInterpretC . hom))) (go m) where
+runInterpret f m = reify (Interpreter (\ hdl sig -> InterpretC . f (runInterpretC . hdl) sig)) (go m) where
   go :: InterpretC s eff m x -> Const (m x) s
   go (InterpretC m) = Const m
 
 -- | Interpret an effect using a higher-order function with some state variable.
 --
 -- @since 1.0.0.0
 runInterpretState
-  :: (forall n x . Monad n => (forall y . n y -> StateC s m y) -> s -> eff n x -> m (s, x))
+  :: (forall ctx n x . Functor ctx => Handler ctx n (StateC s m) -> eff n x -> s -> ctx () -> m (s, ctx x))
```

Comment:
As with the context, the state `s` should follow the signature: `runInterpretState` interprets `eff` into (stateful) arrows modulo `ctx`.